goal_src/engine/camera/cam-master.gc

```diff
@@ -6,4 +6,2664 @@
+    (-> *CAMERA-bank* default-string-min-y)
+    )
+   )
+  (set!
+   (-> self stringMaxHeight)
+   (cam-slave-get-float
+    arg0
+    'stringMaxHeight
+    (-> *CAMERA-bank* default-string-max-y)
+    )
+   )
+  (set!
+   (-> self stringMinLength)
+   (cam-slave-get-float
+    arg0
+    'stringMinLength
+    (-> *CAMERA-bank* default-string-min-z)
+    )
+   )
+  (set!
+   (-> self stringMaxLength)
+   (cam-slave-get-float
+    arg0
+    'stringMaxLength
+    (-> *CAMERA-bank* default-string-max-z)
+    )
+   )
+  (let ((s5-0 (new 'stack-no-clear 'vector)))
+   (when
+    (cam-slave-get-vector-with-offset
+     (the-as entity-actor arg0)
+     s5-0
+     'interesting
+     )
+    (let ((a1-5 (new 'stack-no-clear 'event-message-block)))
+     (set! (-> a1-5 from) self)
+     (set! (-> a1-5 num-params) 1)
+     (set! (-> a1-5 message) 'point-of-interest)
+     (set! (-> a1-5 param 0) (the-as uint s5-0))
+     (send-event-function *camera* a1-5)
+     )
+    )
+   )
+  (if (< 405504.0 (-> self stringMaxLength))
+   (set! (-> self stringMaxLength) (-> *CAMERA-bank* default-string-max-z))
+   )
+  (set!
+   (-> self stringCliffHeight)
+   (cam-slave-get-float arg0 'stringCliffHeight 163840.0)
+   )
+  (let ((a1-7 (new 'stack-no-clear 'event-message-block)))
+   (set! (-> a1-7 from) self)
+   (set! (-> a1-7 num-params) 1)
+   (set! (-> a1-7 message) 'query-state)
+   (set! (-> a1-7 param 0) (the-as uint *camera-base-mode*))
+   (when (not (send-event-function *camera* a1-7))
+    (let ((a1-8 (new 'stack-no-clear 'event-message-block)))
+     (set! (-> a1-8 from) self)
+     (set! (-> a1-8 num-params) 2)
+     (set! (-> a1-8 message) 'change-state)
+     (set! (-> a1-8 param 0) (the-as uint *camera-base-mode*))
+     (set! (-> a1-8 param 1) (the-as uint 450))
+     (send-event-function *camera* a1-8)
+     )
+    )
+   )
+  (when (logtest? #x10000 (cam-slave-get-flags (-> self cam-entity) 'flags))
+   (let ((a1-10 (new 'stack-no-clear 'event-message-block)))
+    (set! (-> a1-10 from) self)
+    (set! (-> a1-10 num-params) 1)
+    (set! (-> a1-10 message) 'set-slave-option)
+    (set! (-> a1-10 param 0) (the-as uint #x10000))
+    (send-event-function *camera* a1-10)
+    )
+   )
+  (let
+   ((f0-12
+     (cam-slave-get-float
+      arg0
+      'tiltAdjust
+      (-> *CAMERA-bank* default-tilt-adjust)
+      )
+     )
+    )
+   (set! (-> *camera-combiner* tracking tilt-adjust target) f0-12)
+   f0-12
+   )
+  )
+
+(defun setup-slave-for-hopefull ((arg0 camera-slave))
+  (when (= (-> arg0 blend-to-type) 2)
+   (cam-calc-follow! (-> arg0 tracking) (-> arg0 trans) #f)
+   (slave-set-rotation!
+    (-> arg0 tracking)
+    (-> arg0 trans)
+    (the-as float (-> arg0 options))
+    (-> arg0 fov)
+    #f
+    )
+   )
+  (none)
+  )
+
+(defbehavior
+  master-is-hopeful-better? camera-master
+  ((arg0 camera-master) (arg1 camera-master))
+  (if (not *camera-combiner*)
+   #f
+   (<
+    (vector-dot
+     (the-as vector (-> arg0 string-max))
```

Comment:
I'm a little suspicious that the argument types here aren't `camera-master` - this cast to a vector is weird.